staging/src/k8s.io/pod-security-admission/policy/checks.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package policy
+
+import (
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Check interface {
+	ID() string
+	// CheckPod determines if the pod is allowed.
+	CheckPod(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) CheckResult
+}
+
+// CheckResult contains the result of checking a pod and indicates whether the pod is allowed,
+// and if not, why it was forbidden.
+//
+// Example output for (false, "host ports", "8080, 9090"):
+//   When checking all pods in a namespace:
+//     disallowed by policy "baseline": host ports, privileged containers, non-default capabilities
+//   When checking an individual pod:
+//     disallowed by policy "baseline": host ports (8080, 9090), privileged containers, non-default capabilities (CAP_NET_RAW)
+type CheckResult struct {
+	// Allowed indicates if the check allowed the pod.
+	Allowed bool
+	// ForbiddenReason should only be set if Allowed is false.
+	// ForbiddenReason should be as succinct as possible and is always output.
+	ForbiddenReason string
```

Comment:
does this field mirror condition Reasons as CamelCase and machine parseable?

Reply:
no, this is a "short version" of the error (e.g. `allowPrivilegeEscalation != false`, `forbidden seLinuxOptions`)